pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -218,7 +226,7 @@ func (a Adapter) Build(parameters common.BuildParameters) (err error) {
 	}
 
 	if isBuildConfigSupported && !parameters.Rootless {
-		return a.runBuildConfig(client, parameters, isImageRegistryInternal)
+		return a.runBuildConfig(true, client, parameters, isImageRegistryInternal)
```

Comment:
This should not be true by default but should come from some isS2i variable based on whether s2i component was present or not (switching logic)